code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,201 @@
+			if("transform")
+				value = appearance:transform
+			if("underlays")
+				value = appearance:underlays
+
+			if("parent_type")
+				value = appearance:parent_type
+			if("type")
+				value = "/appearance (as [appearance:type])" // don't fool people
+
+			// These are not documented ones but trackable values. Maybe we'd want these.
+			if("animate_movement")
+				value = appearance:animate_movement
+			if("dir")
+				value = appearance:dir
+			if("glide_size")
+				value = appearance:glide_size
+			if("pixel_step_size")
+				value = "" //atom_appearance:pixel_step_size
+				// DM compiler complains this
+
+			// I am not sure if these will be ever detected, but I made a connection just in case.
+			if("contents")
+				value = appearance:contents
+			if("vis_contents")
+				value = appearance:vis_contents
+			if("vis_flags") // DM document says /appearance has this, but it throws error
+				value = appearance:vis_flags
+			if("loc")
+				value = appearance:loc
+
+			// we wouldn't need these, but let's these trackable anyway...
+			if("density")
+				value = appearance:density
+			if("screen_loc")
+				value = appearance:screen_loc
+			if("verbs")
+				value = appearance:verbs
+			if("tag")
+				value = appearance:tag
+
+			else
+				return "<li style='backgroundColor:white'>(READ ONLY) [var_name] <font color='blue'>(Undefined var name in switch)</font></li>"
+	catch
+		return "<li style='backgroundColor:white'>(READ ONLY) <font color='blue'>[var_name] = (untrackable)</font></li>"
+	return "<li style='backgroundColor:white'>(READ ONLY) [var_name] = [_debug_variable_value(var_name, value, 0, appearance, sanitize = TRUE, display_flags = NONE)]</li>"
+
+/// Shows a header name on top when you investigate an appearance
+/proc/vv_get_header_appearance(image/thing)
+	. = list()
+	var/icon_name = "<b>[thing.icon || "null"]</b><br/>"
+	. += replacetext(icon_name, "icons/obj", "") // shortens the name. We know the path already.
+	if(thing.icon)
+		. += thing.icon_state ? "\"[thing.icon_state]\"" : "(icon_state = null)"
+
+/image/vv_get_header() // it should redirect to global proc version because /appearance can't call a proc, unless we want dupe code here
+	return vv_get_header_appearance(src)
+
+/// Makes a format name for shortened vv name.
+/proc/get_appearance_vv_summary_name(image/thing)
+	var/icon_file_name = thing.icon ? splittext("[thing.icon]", "/") : "null"
+	if(islist(icon_file_name))
+		icon_file_name = length(icon_file_name) ? icon_file_name[length(icon_file_name)] : "null"
+	if(thing.icon_state)
+		return "[icon_file_name]:[thing.icon_state]"
+	else
+		return "[icon_file_name]"
+
+/proc/vv_get_dropdown_appearance(image/thing)
+	. = list()
+	// unless you have a good reason to add a vv option for /appearance,
+	// /appearance type shouldn't alloow any vv option. Even "Mark Datum" is a questionable behaviour here.
+	VV_DROPDOWN_OPTION_APPEARANCE(thing, "", "---")
+	VV_DROPDOWN_OPTION_APPEARANCE(thing, "", "VV option not allowed")
```

Comment:
```suggestion
	VV_DROPDOWN_OPTION_APPEARANCE(thing, "", "VV option not allowed")
	return .
```

it's not really obvious that we're modifying `.` unless we do this to be honest.